narwhals/_ibis/expr_dt.py

```diff
@@ -91,6 +92,7 @@ def replace_time_zone(self, time_zone: str | None) -> IbisExpr:
             raise NotImplementedError(msg)
 
     nanosecond = not_implemented()
+    offset_by = not_implemented()
```

Comment:
I think it works to just add a `timedelta` for ibis, shall we re-use the same logic from Dask? i.e. just support what can be expressed in terms of timedelta, exclude month/year/nanosecond